node/network/availability-distribution/src/lib.rs

```diff
@@ -0,0 +1,1552 @@
+	}
+
+	const TIMEOUT: Duration = Duration::from_millis(100_000);
+
+	async fn overseer_signal(
+		overseer: &mut test_helpers::TestSubsystemContextHandle<AvailabilityDistributionMessage>,
+		signal: OverseerSignal,
+	) {
+		overseer
+			.send(FromOverseer::Signal(signal))
+			.timeout(TIMEOUT)
+			.await
+			.expect("10ms is more than enough for sending signals.");
+	}
+
+	async fn overseer_send(
+		overseer: &mut test_helpers::TestSubsystemContextHandle<AvailabilityDistributionMessage>,
+		msg: AvailabilityDistributionMessage,
+	) {
+		log::trace!("Sending message:\n{:?}", &msg);
+		overseer
+			.send(FromOverseer::Communication { msg })
+			.timeout(TIMEOUT)
+			.await
+			.expect("10ms is more than enough for sending messages.");
+	}
+
+	async fn overseer_recv(
+		overseer: &mut test_helpers::TestSubsystemContextHandle<AvailabilityDistributionMessage>,
+	) -> AllMessages {
+		log::trace!("Waiting for message ...");
+		let msg = overseer
+			.recv()
+			.timeout(TIMEOUT)
+			.await
+			.expect("TIMEOUT is enough to recv.");
+		log::trace!("Received message:\n{:?}", &msg);
+		msg
+	}
+
+	fn dummy_occupied_core(para: ParaId) -> CoreState {
+		CoreState::Occupied(OccupiedCore {
+			para_id: para,
+			next_up_on_available: None,
+			occupied_since: 0,
+			time_out_at: 5,
+			next_up_on_time_out: None,
+			availability: Default::default(),
+			group_responsible: GroupIndex::from(0),
+		})
+	}
+
+	fn generate_n_validators(n: usize) -> Vec<ValidatorId> {
+		iter::repeat(n)
+			.take(n)
+			.map(|_| {
+				let validator_pair = ValidatorPair::generate();
+				validator_pair.0.public()
+			})
+			.collect()
+	}
+
+	fn valid_availability_gossip(
+		candidate_hash: H256,
+		validator_count: usize,
+		validator_index: ValidatorIndex,
+		erasure_chunk_index: u32,
+	) -> AvailabilityGossipMessage {
+		let available_data = AvailableData {
+			pov: PoV {
+				block_data: BlockData(vec![0x77; 453]),
+			},
+			omitted_validation: OmittedValidationData {
+				global_validation: GlobalValidationData::default(),
+				local_validation: LocalValidationData {
+					validation_code_hash: candidate_hash.clone(),
+					..Default::default()
+				},
+			},
+		};
+		let erasure_chunks =
+			derive_erasure_chunks_with_proofs(validator_count, validator_index, &available_data)
+				.expect("Generated must work");
+
+		let erasure_chunk: ErasureChunk = erasure_chunks
+			.get(erasure_chunk_index as usize)
+			.expect("Must be valid or input is oob")
+			.clone();
+
+		AvailabilityGossipMessage {
+			candidate_hash,
+			erasure_chunk,
+		}
+	}
+
+	#[test]
+	fn reputation_verification() {
+		let _ = env_logger::builder()
+			.is_test(true)
+			.filter(None, log::LevelFilter::Trace)
+			.try_init();
+		let keystore = keystore::Store::new_in_memory();
+
+		let candidate_hash_af = H256::from_slice(&[0x0Fu8; 32]);
+		let candidate_hash_11 = H256::from_slice(&[0x11u8; 32]);
+
+		// @todo fails with invalid seed
+		// let validator_keypair: ValidatorPair = keystore
```

Comment:
Try `Sr25519Keyring` - you can see examples in other subsystem crates or in the runtime

`Sr25519Keyring::Alice.pair().into()` should work